packages/starknet-snap/src/types/snapState.ts

```diff
@@ -134,4 +143,26 @@ export type Transaction = {
   timestamp: number;
 };
 
+export type V2Transaction = {
+  txnHash: string; // in hex
+  txnType: StarkNetTransactionType;
+  chainId: string; // in hex
+  senderAddress: string; // in hex
+  contractAddress: string; // in hex
+  executionStatus?: TransactionExecutionStatus | string;
+  finalityStatus?: TransactionFinalityStatus | string;
+  failureReason: string;
+  timestamp: number;
+  maxFee?: string | null;
+  actualFee?: string | null;
+  // using Record<string, TranscationAccountCall[]> to support O(1) searching
+  accountCalls?: Record<string, TranscationAccountCall[]> | null;
+  version: 'V2';
```

Comment:
What about V3 transactions ? 

https://starkscan.readme.io/reference/transaction-object

Reply:
it is not for TX v3, it is for separate the type of the state data